src/java.base/share/classes/jdk/crac/Core.java

```diff
@@ -50,6 +52,9 @@
 
     private static boolean traceStartupTime;
 
+    private static final List<CheckpointLock> lockQ = new LinkedList<CheckpointLock>();
```

Comment:
LinkedList is rarely the class we actually want to use.  Better to use `ArrayList`

Reply:
Thank you. After comparing ArrayList and LinkedList implementations, I can see - ArrayList has better performance even for general LinkedList use cases.